### PR TITLE
Ensure all <mwc-fab> set a label for ARIA

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

`<mwc-fab>` uses the `label` attribute to set the `aria-label` for its included elements (= the actual button). This PR ensure that we set it for all `<mwc-fab>`.

https://npm.io/package/@material/mwc-fab

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/frontend/issues/3267
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/areas/ha-config-areas-dashboard.ts [hunk 1]
```diff
@@ -129,6 +129,9 @@ export class HaConfigAreasDashboard extends LitElement {
           title="${this.hass.localize(
             "ui.panel.config.areas.picker.create_area"
           )}"
+          label="${this.hass.localize(
```
Comment (bramkragten) [inline src/panels/config/areas/ha-config-areas-dashboard.ts:132]:
```suggestion
          .label=${this.hass.localize(
```

File: src/panels/config/automation/ha-automation-editor.ts [hunk 2]
```diff
@@ -473,6 +473,7 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
           slot="fab"
           class=${classMap({ dirty: this._dirty })}
           .title=${this.hass.localize("ui.panel.config.automation.editor.save")}
+          label=${this.hass.localize("ui.panel.config.automation.editor.save")}
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:476]:
```suggestion
          .label=${this.hass.localize("ui.panel.config.automation.editor.save")}
```

File: src/panels/config/automation/ha-automation-picker.ts [hunk 3]
```diff
@@ -179,6 +179,9 @@ class HaAutomationPicker extends LitElement {
           title=${this.hass.localize(
             "ui.panel.config.automation.picker.add_automation"
           )}
+          label=${this.hass.localize(
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-picker.ts:182]:
```suggestion
          .label=${this.hass.localize(
```

File: src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts [hunk 4]
```diff
@@ -85,6 +85,7 @@ class ZHAConfigDashboard extends LitElement {
         <a href="/config/zha/add" slot="fab">
           <mwc-fab
             title=${this.hass.localize("ui.panel.config.zha.add_device")}
+            label=${this.hass.localize("ui.panel.config.zha.add_device")}
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts:88]: We should not have a title and a label

**Comment by bramkragten** [comment]
When you set a dynamic value (like a translation) use the property (with the `.`) to set it instead of the attribute.

File: src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts (hunk 4 above)
Comment (spacegaier) [reply src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts:88]: But then we either miss the tooltip or the internal aria-label support for screen readers:
- `title` is used to set the tooltip
- `label` is used by mwc for aria labels

To limit the number of strings we pass around we would need to create something like `HaFab` where we only pass one property and then use it internally to set both `title` and `label`.

Comment (spacegaier) [reply src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts:88]: If however we combine my changes with the ones from https://github.com/home-assistant/frontend/pull/7612 we would not really need a tooltip anymore if we always use the `extended` mode = show a text in the frontend.

**Comment by spacegaier** [comment]
@adamjernst FYI: I combined your proposed changes from https://github.com/home-assistant/frontend/pull/7612 into this PR since I anyway touched basically all `<mwc-fab>` here already.

**Comment by adamjernst** [comment]
> @adamjernst FYI: I combined your proposed changes from #7612 into this PR since I anyway touched basically all `<mwc-fab>` here already.

Beautiful, thanks! I'll close my PRs as soon as this is merged.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**